pkg/services/cloudprovider/csi.go

```diff
@@ -437,10 +435,7 @@ func CSIControllerDeployment(storageConfig *v1alpha3.CPIStorageConfig) *appsv1.D
 						{
 							Name: "socket-dir",
 							VolumeSource: corev1.VolumeSource{
-								HostPath: &corev1.HostPathVolumeSource{
-									Path: "/var/lib/csi/sockets/pluginproxy/csi.vsphere.vmware.com",
-									Type: newHostPathType(string(corev1.HostPathDirectoryOrCreate)),
```

Comment:
can you check the container flags, we're missing for example things like https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/release-2.1/manifests/v2.1.0/vsphere-67u3/vanilla/deploy/vsphere-csi-controller-deployment.yaml#L105 and I suspect other things too

Reply:
updated the args and the clusterrole